tests/test_volume.py

```diff
@@ -0,0 +1,540 @@
+    create_volume(vol_name1)
+    try:
+        # now try creating the same volume again
+        create_volume(vol_name1)
+    except exceptions.NimOSAPIError as ex:
+        # covered SM_eexist and SM_http_conflict
+        if 'SM_eexist' in str(ex):
+            log(f"Failed As Expected")
+    except Exception as exgeneral:
+        log(exgeneral)
+
+
+@pytest.mark.skipif(SKIPTEST is True,
+                    reason="skipped this test as SKIPTEST variable is true")
+def test_create_volume_bad_params(setup_teardown_for_each_test):
+    # , or / is not supported by nimble for creating vol name
+    vol_name = vol_name1 + "/,"
+    try:
+        create_volume(vol_name)
+    except exceptions.NimOSAPIError as ex:
+        # protocol ex is SM_http_bad_request
+        if 'SM_invalid_arg_value' in str(ex):
+            log(f"Failed As Expected. Invalid vol_name to create: {vol_name}")
+
+
+@pytest.mark.skipif(SKIPTEST is True,
+                    reason="skipped this test as SKIPTEST variable is true")
+def test_create_volume_unexpected_arg(setup_teardown_for_each_test):
+    try:
+        # "invalidarg" is not a part of volume argument.
+        nimosclientbase.get_nimos_client().volumes.create(
+            vol_name1, size=50, invalidarg="testinvalidarg")
+    except exceptions.NimOSAPIError as ex:
+        # protocol ex is SM_http_bad_request
+        if 'SM_unexpected_arg' in str(ex):
+            log("Failed As Expected. "
+                f"Unexpected arg for vol_name : {vol_name1}")
+
+
+@pytest.mark.skipif(SKIPTEST is True,
+                    reason="skipped this test as SKIPTEST variable is true")
+def test_volume_page_size(setup_teardown_for_each_test):
+    resp_vol = []
+    # first atleast create few volume
+    for i in range(0, 5):
+        vol_name = nimosclientbase.get_unique_string("volumetc-vol1-" + str(i))
+        resp = create_volume(vol_name)
+        resp_vol.append(resp)
+    resp = nimosclientbase.get_nimos_client().volumes.list(
+        detail=True, pageSize=2)
+    assert resp is not None
+    assert resp.__len__() == 2
+
+
+@pytest.mark.skipif(SKIPTEST is True,
+                    reason="skipped this test as SKIPTEST variable is true")
+def test_invalid_volume_page_size(setup_teardown_for_each_test):
+    # first atleast create few volume
+    for i in range(0, 6):
+        vol_name = nimosclientbase.get_unique_string("volumetc-vol1-" + str(i))
+        create_volume(vol_name)
+    try:
+        nimosclientbase.get_nimos_client().volumes.list(
+                detail=True, pageSize=5000)
+    except Exception as ex:
+        if"SM_too_large_page_size" in str(ex):
+            log("Failed as expected. Invaild pagesize given")
+        else:
+            log(ex)
+
+
+@pytest.mark.skipif(SKIPTEST is True,
+                    reason="skipped this test as SKIPTEST variable is true")
+def test_create_volume_readonly(setup_teardown_for_each_test):
+    # create a read only volume..the volume gets created as write only.
```

Comment:
only volume.  The volume gets created as read/write

Reply:
But the volume gets created as write only.